### ISSUE TITLE
Compile Error on New Core

### ISSUE THREAD

**Issue by lbussy** [lbussy/LCBUrl#27]
After latest upgrades:

```
platformio/espressif8266 -> 3.0.0
platformio/toolchain-xtensa -> 2.100200.0 @ ~2.100200.0
platformio/framework-arduinoespressif8266 -> 3.30000.210519 @ ~3.30000.0
```

The compiler complains about the way I'm doing some string/char manipulations:

```
.pio\libdeps\d1_mini\LCBUrl\src\LCBUrl.cpp: In member function 'bool LCBUrl::isMDNS(const char*)':
.pio\libdeps\d1_mini\LCBUrl\src\LCBUrl.cpp:579:21: error: invalid conversion from 'char' to 'char*' [-fpermissive]
  579 |  char * lastLabel = '\0';
      |                     ^~~~
      |                     |
      |                     char
```

**Comment by coraxium** [comment]
Possible solution - replace the line
```
	char * lastLabel = '\0';`
```
to
```
	char * lastLabel;
	*lastLabel = '\0';
```
and it does compile.

**Comment by lbussy** [comment]
I'd forgotten all about this!

**Comment by lbussy** [comment]
As I suspected, I'd fixed this and it's still languishing in `new_core`:

https://github.com/lbussy/LCBUrl/blob/e32fd31d9d66259175522580f7642a68471d840f/src/LCBUrl.cpp#L587

**Comment by coraxium** [comment]
I will give it a try :-)

**Comment by coraxium** [comment]
Solution does run on my machine with compilers from 2.2.1 (legacy) to Version 3

**Comment by lbussy** [comment]
Grumble ... need to fix this and I can't find my esp8266:

```
src/LCBUrl.cpp: In member function 'IPAddress LCBUrl::getIP(const char*)':
src/LCBUrl.cpp:746:9: error: 'esp_err_t' was not declared in this scope
  746 |         esp_err_t result = mdns_query(fqdn);
      |         ^~~~~~~~~
src/LCBUrl.cpp:747:13: error: 'n' was not declared in this scope; did you mean 'yn'?
  747 |         if (n == 0)
      |             ^
      |             yn
*** [.pio/build/d1_mini/src/LCBUrl.cpp.o] Error 1
```
Something changed with mDNS resolution.

**Comment by thorrak** [comment]
This should be fixed by #31
